ci/docs-builder.py

```diff
@@ -0,0 +1,158 @@
+
+
+def load_config(config_name):
+    # Get the config
+    config_file = os.path.join(os.path.dirname(__file__),
+                               'config', 'releases', '%s.yaml' % config_name)
+    if not os.path.exists(config_file):
+        print "Error: %s not found. " % config_file
+        sys.exit(1)
+    with open(config_file, 'r') as config_handle:
+        config = yaml.safe_load(config_handle)
+    return config
+
+def main():
+    # Parse the args
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--release", required=True, help="Build the docs for a given release.")
+    opts = parser.parse_args()
+
+    configuration = load_config(opts.release)
+
+    # Get platform build version
+    repo_list = configuration['repositories']
+    try:
+        pulp_dict = filter(lambda x: x['name'] == 'pulp', repo_list)[0]
+    except IndexError:
+        raise RuntimeError("config file does not have an entry for 'pulp'")
+    version = pulp_dict['version']
+
+    if version.endswith('alpha'):
+        build_type = 'nightly'
+    elif version.endswith('beta'):
+        build_type = 'dev'
+    elif version.endswith('rc'):
+        build_type = 'dev'
+    else:
+        build_type = 'ga'
+
+    x_y_version = '.'.join(version.split('.')[:2])
+
+    builder.ensure_dir(WORKING_DIR, clean=True)
+
+    print "Getting git repos"
+    for component in get_components(configuration):
+        #clone the repos
+        branch_name = component['git_branch']
+        print "Cloning from github: %s" % component.get('git_url')
+        print "Switching to branch %s" % branch_name
+        clone_command = ['git', 'clone', component.get('git_url'), '--branch', branch_name]
+        exit_code = subprocess.call(clone_command, cwd=WORKING_DIR)
+        if exit_code != 0:
+            raise RuntimeError('An error occurred while cloning the repo.')
+
+    plugins_dir = WORKING_DIR + os.sep + 'pulp' + os.sep + 'docs' + os.sep + 'plugins'
+    builder.ensure_dir(plugins_dir)
+
+    for component in get_components(configuration):
+        if component['name'] == 'pulp':
+            continue
+
+        src = WORKING_DIR + os.sep + component['name'] + os.sep + 'docs'
+        dst = plugins_dir + os.sep + component['name']
+        os.symlink(src, dst)
+
+    # copy in the pulp_index.rst file
+    if x_y_version == '2.8' and build_type == 'ga':
+        src_path = 'docs/pulp_index_2_8.rst'
+    else:
+        src_path = 'docs/pulp_index.rst'
+    pulp_index_rst = WORKING_DIR + os.sep + 'pulp' + os.sep + 'docs' + os.sep + 'index.rst'
+    copyfile(src_path, pulp_index_rst)
+
+    # copy in the plugin_index.rst file
+    plugin_index_rst = plugins_dir + os.sep + 'index.rst'
+    copyfile('docs/plugin_index.rst', plugin_index_rst)
+
+    # copy in the all_content_index.rst file
+    all_content_index_rst = WORKING_DIR + os.sep + 'pulp' + os.sep + 'docs' + os.sep + 'all_content_index.rst'
+    copyfile('docs/all_content_index.rst', all_content_index_rst)
```

Comment:
I'd be sorely tempted to crush all these copyfiles down into a loop, like with `destination: source` pairs in a dict, but appreciate the readability of having them all explicitly written out, particularly in a piece of build automation, since it makes it easier to see exactly what's happening.